src/Sdk/DTLogging/Logging/ValueEncoders.cs

```diff
@@ -80,6 +81,65 @@ public static String TrimDoubleQuotes(String value)
             return trimmed;
         }
 
+        public static String PowerShellPreAmpersandPlus(String value)
+        {
+            // if the secret is passed to PS as a command and it causes an error, sections of it can be surrounded by color codes
+            // or printed individually. 
+            
+            // The secret secretpart1&secretpart2&secretpart3 would be split into 2 sections:
+            // 'secretpart1&secretpart2&' and 'secretpart3'. This method masks for the first section.
+            
+            // The secret secretpart1&+secretpart2&secretpart3 would be split into 2 sections:
```

Comment:
great comments 👍

Reply:
Is this behavior the same for powershell and powershell core (pwsh)

Reply:
When I tested it in powershell, it used a single color (red) for the entire error message and split the string the same way.